src/structures/APIMessage.js

```diff
@@ -182,19 +169,12 @@ class APIMessage {
       typeof this.options.allowedMentions === 'undefined'
         ? this.target.client.options.allowedMentions
         : this.options.allowedMentions;
-    if (this.options.reply) {
-      const id = this.target.client.users.resolveID(this.options.reply);
-      if (allowedMentions) {
-        // Clone the object as not to alter the ClientOptions object
-        allowedMentions = Util.cloneObject(allowedMentions);
-        const parsed = allowedMentions.parse && allowedMentions.parse.includes('users');
-        // Check if the mention won't be parsed, and isn't supplied in `users`
-        if (!parsed && !(allowedMentions.users && allowedMentions.users.includes(id))) {
-          if (!allowedMentions.users) allowedMentions.users = [];
-          allowedMentions.users.push(id);
-        }
-      } else {
-        allowedMentions = { users: [id] };
+
+    let message_reference;
+    if (typeof this.options.replyTo !== 'undefined') {
+      const message_id = this.target.messages.resolveID(this.options.replyTo);
+      if (message_id) {
+        message_reference = { message_id };
```

Comment:
I think it would be add channel_id and guild_id.
https://discord.com/developers/docs/resources/channel#message-object-message-reference-structure

Reply:
These fields are not required when sending an inline reply as they can only be to messages in the same channel.

Reply:
if they can only use same channel,
It should be check TargetMessage and replyToMessage in same channel.

Reply:
That's not something Discord.js should be responsible for checking, nor can it without attempting to fetch the message from that channel with the ID provided. This will result in throwing a DiscordAPIError if the message ID is invalid.

Developers will receive alreaey receive a DiscordAPIError error anyway if they attempt to target an invalid message, and can catch that and handle accordingly.